change getter and setter by lombok dependency

deepakbarajati opened the pull request:
via use of lombok dependency we can reduce some code like getter , setter ,allArgConstructor

stefanhahmann commented:
@deepakbarajati 

Thanks for using this repo and even more for contributing to it.
I was aware of the existence of lombok but never really used it myself. I agree that it would simplify the code.
However, as far as I know lombok is not shipped with every IDE and thus plugins need to be installed potentially.

* I would appreciate, if you could add a statement regarding this in https://github.com/FraunhoferIVI/openlr/blob/master/SETUP.md (section IDE)

* Since in this repo there is no automatic test pipeline, did you make sure, if the existing unit tests are still running with your changes?

deepakbarajati commented:
@stefanhahmann
When I try to run existing unit tests, I encounter issues with IncidentsJsonParserTest.java and FlowJsonParserTest.java. These tests use IncidentsJsonParser.java and FlowJsonParser.java respectively, but they are trying to use constructors with no arguments, whereas these classes define constructors that require a HereDecoder argument.

stefanhahmann commented:
> @stefanhahmann When I try to run existing unit tests, I encounter issues with IncidentsJsonParserTest.java and FlowJsonParserTest.java. These tests use IncidentsJsonParser.java and FlowJsonParser.java respectively, but they are trying to use constructors with no arguments, whereas these classes define constructors that require a HereDecoder argument.

Yes, this actually correct. Apologies for this. This repo is unfortunately not in very well maintained state. 

I just made 2 commits to the master branch:

- https://github.com/FraunhoferIVI/openlr/commit/73f44e8f826586eedeb9cd404d8f6f09ba1cb8c4
- https://github.com/FraunhoferIVI/openlr/commit/a59b3f2d0e5f14aefa721b91aa56097c2c57d7c6

They should ensure that it is at least possible to run all existing unit tests in the master branch.

Please merge these commits into your branch and check, if all unit tests are running.

deepakbarajati commented:
@stefanhahmann 
 I am facing an issue with setting up code on both Windows and Ubuntu in IntelliJ IDEA. My IDE is not resolving ** JOOQ ** dependencies. I also tried fixing the issue using the command  **mvn clean compile** , but it gave an authentication error saying that the username and password were not authenticated. I changed the username and password for PostgreSQL locally, but the issue persists. I also changed the username and password in  **DatasourceConfig.java** , but it did not resolve the issue.

stefanhahmann commented:
> @stefanhahmann I am facing an issue with setting up code on both Windows and Ubuntu in IntelliJ IDEA. My IDE is not resolving ** JOOQ ** dependencies. I also tried fixing the issue using the command **mvn clean compile** , but it gave an authentication error saying that the username and password were not authenticated. I changed the username and password for PostgreSQL locally, but the issue persists. I also changed the username and password in **DatasourceConfig.java** , but it did not resolve the issue.

This is unfortunately, the most difficult step of the whole setup process. I am really not happy with it. Many users experience difficulties at this step.
Since you are not mentioning the `pom.xml`, did you see that the `https://github.com/FraunhoferIVI/openlr/blob/master/SETUP.md` instructions mention that the postgresql credential need to be entered, both in `DatasourceConfig.java` (as you already did) *and* in the `pom.xml`?

Cf.: 
![grafik](https://github.com/user-attachments/assets/90ccccb6-e651-4a02-b61d-ff9340a381a1)

deepakbarajati commented:
After making changes to the pom.xml file, including updating the username, database name, and password, as well as modifying the dependencies and Maven plugin, it worked properly and executed successfully

stefanhahmann commented:
> After making changes to the pom.xml file, including updating the username, database name, and password, as well as modifying the dependencies and Maven plugin, it worked properly and executed successfully

Great. If you could confirm that `mvn clean test` runs successfully in your branch, I would merge your changes into the main project. Thanks for testing and improving.

deepakbarajati commented:
**mvn clean test** ran successfully. If you would like to merge, you can.
